### PR TITLE
Renee updates

### PR DESCRIPTION
**Pull request by olson59**
Changes:

- Pyranda is now dumping mixedness, "mixedness is a common metric for mix modeling that is based on the variance"
- I created two versions of the LC yaml. One for generating simulation data to train the GP model, and one for generating fake experimental data at a higher resolution and including random noise perturbations in the initial conditions. This can be generalized to the AWS yaml.
- There is a csv file for Xiao of the simulation data to train the GP's in Psuade. 200+ samples by 4 columns. The first 3 columns are inputs ['atwood', 'velocity', 'time'], and the last column is the output 'mix_width'. 
- There are two pgen files. simulation_pgen.py creates samples with a Latinhypercube design, and experimental_pgen.py creates samples at a higher resolution with noise. 
- plot_study.py is useful to create plots to view the data and fit a GP.

### CODE REVIEW

File: CZ/pyranda_rayleigh_taylor/README.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Rayleigh Taylor Modeling with pyranda
+
+pyranda is an mpi parallel high order finited difference solver for arbitrary hyperbolic PDE systems (INSERT LINK).  This set of examples works through simulations of the Rayleigh-Taylor problem,
```
Comment (bgunnar5) [inline CZ/pyranda_rayleigh_taylor/README.md]: Do we know which link we need to insert here yet?

File: CZ/pyranda_rayleigh_taylor/bayesian_inference.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+seed = 15
+surrogate = GaussianProcessRegressor().fit(X, Y[:, i])
+ranges = [[0, 1], [0, 1], [0, 1]]
+flattened = False
+scaled = True
+fileName = "mytestfile.hdf5"
+with h5py.File(fileName, "w") as f:
+    f.create_dataset("inputs", data=X)
+    f.create_dataset("outputs", data=Y)
+    f.close()
```
Comment (bgunnar5) [inline CZ/pyranda_rayleigh_taylor/bayesian_inference.py]: I don't think `f.close()` is necessary if we're using the `with h5py.File(..)` context manager (see [Closing Files](https://docs.h5py.org/en/stable/high/file.html#closing-files))

File: CZ/pyranda_rayleigh_taylor/bayesian_inference.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+dataset = store.create("uq_data")
+dataset.associate([fileName], 'hdf5')
+
+result = kosh_operators.KoshMCMC(dataset["inputs"],
+                                 method="default_mcmc",
+                                 input_names=input_names,
+                                 inputs_low=[0.0]*3,
+                                 inputs_high=[1.0]*3,
+                                 proposal_sigmas=[0.2]*3,
+                                 prior=,
```
Comment (bgunnar5) [inline CZ/pyranda_rayleigh_taylor/bayesian_inference.py]: I'm not familiar with this setting, is it purposefully left blank?

File: CZ/pyranda_rayleigh_taylor/experimental_pgen.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
+import numpy as np
+from maestrowf.datastructures.core import ParameterGenerator
+
+# Generate sample points for the parameters from normal distributions
+# for our fake experimental data
+Nexperiments = 8
```
Comment (bgunnar5) [inline CZ/pyranda_rayleigh_taylor/experimental_pgen.py]: Should we make this a variable in the Maestro spec that we can vary? Variables from the `env` block can be read into the pgen file (instructions for how this is done can be found [here](https://maestrowf.readthedocs.io/en/latest/Maestro/parameter_specification.html#referencing-values-from-a-specifications-env-block)).

File: CZ/pyranda_rayleigh_taylor/plot_study.py [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+import glob
+import os
+from sklearn.preprocessing import MinMaxScaler as MMS
+from sklearn.gaussian_process import GaussianProcessRegressor as GPR
+
+rng = np.random.default_rng()
+
+# This is single mode cases (too easy for GP's)
+# dataDir = "RT_STUDIES/rayleigh_taylor_pyranda_20240813-113206/run-pyranda/*"
+dataDir = "RT_STUDIES/rayleigh_taylor_pyranda_20240813-125700/run-pyranda/*"
```
Comment (bgunnar5) [inline CZ/pyranda_rayleigh_taylor/plot_study.py]: Should this be an argument that the user can pass into this script that way it's not hardcoded in?

File: CZ/pyranda_rayleigh_taylor/rayleigh_taylor.py [hunk 6]
```diff
@@ -0,0 +1,646 @@
...
+        ydom = (0.0, 2*pi*FF,  Npts, periodic=True)
+        zdom = (0.0, 2*pi*FF,  Npts, periodic=True)
+        """.replace(
+                "Npts", str(Npts)
+            )
+            .replace("pi", str(numpy.pi))
+            .replace("FF", str(float(Npts - 1) / Npts))
+            .replace("xwidth", str(xwidth))
+        )
+        waveLength = 4
```
Comment (bgunnar5) [inline CZ/pyranda_rayleigh_taylor/rayleigh_taylor.py]: Wondering if we should just change this to a single string for imesh that's located outside of the if/else since they're so similar. Something like:
```python
if is2D:
    widthToHeight = 4.0
    xwidth = widthToHeight*numpy.pi
    problem = "RAYLEIGH_TAYLOR_2D"
    waveLength = 4
    zpts = 1
else:
    widthToHeight = 3.0
    xwidth = widthToHeight*numpy.pi
    problem = "RAYLEIGH_TAYLOR_3D"
    waveLength = 4
    zpts = Npts

imesh = (
    """
    xdom = (-xwidth/2.0, xwidth/2.0, int(Npts*widthToHeight), periodic=False)
    ydom = (0.0, 2*pi*FF,  Npts, periodic=True)
    zdom = (0.0, 2*pi*FF,  zpts, periodic=True)
    """.replace(
            "Npts", str(Npts)
        )
        .replace("pi", str(numpy.pi))
        .replace("FF", str(float(Npts - 1) / Npts))
        .replace("xwidth", str(xwidth))
        .replace("widthToHeight",str(widthToHeight))
        .replace("zpts",str(zpts))
    )
```

**Review by doutriaux1 — APPROVED** [review]
Awesome. Thx Renee

**Review by bgunnar5 — COMMENTED** [review]
Looks great, thanks for the work on this Renee! I just had a few small comments on some things here, nothing major.

File: CZ/pyranda_rayleigh_taylor/bayesian_inference.py (hunk 3 above)
Comment (olson59) [reply CZ/pyranda_rayleigh_taylor/bayesian_inference.py]: The default is a uniform distribution and I need to figure out what to put there to trigger the default, or I can just provide uniform myself. This bayesian inference file is not done yet.

File: CZ/pyranda_rayleigh_taylor/README.md (hunk 1 above)
Comment (olson59) [reply CZ/pyranda_rayleigh_taylor/README.md]: https://github.com/LLNL/pyranda
Are you looking for something like this?

Comment (bgunnar5) [reply CZ/pyranda_rayleigh_taylor/README.md]: Yeah I think linking to the pyranda project itself would work.

File: CZ/pyranda_rayleigh_taylor/bayesian_inference.py (hunk 3 above)
Comment (bgunnar5) [reply CZ/pyranda_rayleigh_taylor/bayesian_inference.py]: Ah ok, sorry for jumping the gun on this then.

File: CZ/pyranda_rayleigh_taylor/README.md (hunk 1 above)
Comment (jwhite242) [reply CZ/pyranda_rayleigh_taylor/README.md]: also, no need to fix that in this one: that's a mess i put in there, lol.  the whole readme needs a bit more filling out on the nominal behaviors like some snapshots/curves and stuff too to give a bit of a view of it without running everything

File: CZ/pyranda_rayleigh_taylor/rayleigh_taylor_lc_build_venv_sim.yaml [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+
+batch:
+    type        : slurm
+    host        : rzwhippet
+    bank        : wbronze
+    queue       : pdebug
+
+
+study:
+    # - name: make-env-lc
```
Comment (jwhite242) [inline CZ/pyranda_rayleigh_taylor/rayleigh_taylor_lc_build_venv_sim.yaml]: want to uncomment these make-env steps given they're the 'build_venv' variants of the studies?

**Review by jwhite242 — APPROVED** [review]
All looks pretty good.  Just a few over arching comments, which don't necessarily need fixing right now:

* for the eventual uq variants we probably want to change the study names and descriptions as well, and maybe even output_path variables?
* for the longer term (and this really doesn't matter for the coming aws tutorials given we're building the venv up front), I think we'll want some requirements files for all these various options that enables building and running things from scratch.  also a good place limit the package versions to upstream changes are less likely to break things for users without us knowing
